### ISSUE TITLE
Malformated HTML tags cause parsing error

### ISSUE THREAD

**Issue by Goaman** [odoo-dev/jabberwock#51]
Initialize the editor with:
```html
<p><i>a</p> <p>b</i></p>
```

Will render:
```html
<p><i>a</i></p><i>&nbsp;</i><p><i>b</i></p>
```
But if we place the `<i>` tag properly:
```html
<i><p>a</p> <p>b</p></i>
```

It render:
```html
<p><i>a</i></p><p><i>b</i></p>
```

**Comment by dmo-odoo** [comment]
Are you sure having a `p` inside a `i` is valid? I would not be surprised if the browser itself already corrects it into `<p><i>a</i></p><p><i>b</i></p>` on `innerHTML` so that part might not be a problem of our parser.

**Comment by Goaman** [comment]
You are right, as the spec specify (https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-i-element). `<i>` element cannot exist outside a "phrasing context".

**Comment by dmo-odoo** [comment]
The nbsp is strange though, as they were not added by the browser when I tested it, so it looks like it might be our code that does that, which is surprising honestly...

**Comment by dmo-odoo** [comment]
Is this issue still accurate as of the current `master` ?

**Comment by Goaman** [comment]
Yes.
